src/hints/mod.rs

```diff
@@ -259,9 +277,87 @@ pub fn load_next_tx(
     ap_tracking: &ApTracking,
     _constants: &HashMap<String, Felt252>,
 ) -> Result<(), HintError> {
-    let mut transactions = exec_scopes.get::<Iter<InternalTransaction>>("transactions")?;
+    let mut transactions = exec_scopes.get::<IntoIter<InternalTransaction>>("transactions")?;
     // Safe to unwrap because the remaining number of txs is checked in the cairo code.
     let tx = transactions.next().unwrap();
     exec_scopes.insert_value("transactions", transactions);
+    exec_scopes.insert_value("tx", tx.clone());
     insert_value_from_var_name("tx_type", Felt252::from_bytes_be(tx.r#type.as_bytes()), vm, ids_data, ap_tracking)
 }
+
+/// Implements hint:
+///
+/// ids.contract_address_salt = tx.contract_address_salt
+/// ids.class_hash = tx.class_hash
+/// ids.constructor_calldata_size = len(tx.constructor_calldata)
+/// ids.constructor_calldata = segments.gen_arg(arg=tx.constructor_calldata)
+pub fn prepare_constructor_execution(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    // Safe to unwrap because the remaining number of txs is checked in the cairo code.
+    insert_value_from_var_name(
+        "contract_address_salt",
+        tx.contract_address_salt.unwrap_or(Felt252::from(0)),
+        vm,
+        ids_data,
+        ap_tracking,
+    )?;
+    insert_value_from_var_name("class_hash", tx.class_hash.unwrap_or(Felt252::from(0)), vm, ids_data, ap_tracking)?;
+
+    let constructor_calldata_size = match &tx.constructor_calldata {
+        None => 0,
+        Some(calldata) => calldata.len(),
+    };
+    insert_value_from_var_name("constructor_calldata_size", constructor_calldata_size, vm, ids_data, ap_tracking)?;
+
+    let constructor_calldata: Vec<MaybeRelocatable> = match tx.constructor_calldata {
+        None => vec![],
+        Some(calldata) => calldata.iter().map(|felt| felt.into()).collect(),
+    };
```

Comment:
Is that needed? Can't you just do `calldata.iter().map(|felt| felt.into()).collect()` and it'd be an empty vec if the length is 0?

Reply:
I am not sure what you meant here, if you were talking about removing `match` part then I can't do that because it's a `Option`. So I can't `iter` directly. However, I did modify the code to make it smaller and use `unwrap_or_default`